jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/RegexPathSpec.java

```diff
@@ -193,4 +194,21 @@ public boolean matches(final String path)
             return getMatcher(path).matches();
         }
     }
+
+    @Override
+    public boolean equals(Object o)
```

Comment:
This code is duplicated 3 times, why don't we have a simple base class like `AbstractPathSpec`, which could implement this and also maybe `compareTo` and `toString`, and then we could reduce the code duplication.

I made some similar changes like this in my branch at `jetty-9.4.x-4877-PathSpecEquals` if you want to take a look
https://github.com/eclipse/jetty.project/blob/c6a40782571a08ae2410c165bfc58da5976432c8/jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/AbstractPathSpec.java#L23-L60

Reply:
@lachlan-roberts because I asked him to make all the fields final. Hard to do the abstract and the final fields without more complication.

Reply:
@gregw Take a look at the file I linked, it doesn't prevent having any final fields in the implementations of the abstract class.

Reply:
I like the idea of this abstract class that provides implementations for `toString`, `equals`, `hashcode` and `compareTo`, especially if I can get rid of the  `PathSpec.compareTo` default implementation. But I'd make `equals` and `hashcode` final then to make sure `compareTo` and `equals` always agree. See [my comment](https://github.com/eclipse/jetty.project/issues/4877#issuecomment-638171053) for context.